Source/WebCore/rendering/style/RenderStyle.h

```diff
@@ -515,6 +515,8 @@ class RenderStyle {
     inline OverscrollBehavior overscrollBehaviorY() const;
     
     Visibility visibility() const { return static_cast<Visibility>(m_inheritedFlags.visibility); }
+    inline Visibility visibilityForPainting() const;
```

Comment:
`usedVisibility()` is the typical naming scheme we use for this.

Reply:
If it's not strictly "used" I think "for painting" is good.

Reply:
Why does hit-testing follow specified visibility? It seems weird to be able to click on invisible things.

Reply:
If those cases are really obscure then `usedVisibility()` would seem like the most consistent choice.

Reply:
> Why does hit-testing follow specified visibility?

Actually, now that I think more about it — we can probably use the `visibilityForPainting` / `usedVisibility` for hit-testing by default, but add a special option in the few places where we need to honor the original, specified visibility. If we did this refactoring, then I agree that `usedVisibility` would be an appropriate name. I'll give this a try, thanks!